gwdetchar/utils.py

```diff
@@ -121,7 +121,10 @@ def row(seg, channel):
 
     for name, flag in flagdict.items():
         rows.extend(map(partial(row, channel=name), flag.active))
-    table = EventTable(rows=rows, names=names)
+    if rows:
+        table = EventTable(rows=rows, names=names)
+    else:
+        table = EventTable(names=names)
```

Comment:
You _could_ simplify this to

```python
    table = EventTable(rows=rows or None, names=names)
```

and remove the conditional, but the current way is probably more readable.

Reply:
Thanks @duncanmmacleod, I like the way your suggestion takes up fewer lines, so I went ahead and implemented it.